aten/src/ATen/native/QuantizedLinear.cpp

```diff
@@ -46,125 +45,120 @@ Tensor fbgemm_linear_int8_weight_fp32_activation(
   TORCH_CHECK(fbgemm::fbgemmSupportedCPU(), "Your CPU doesn't support FBGEMM.");
 
-  auto input_contig = input.contiguous();
-  auto* input_ptr = input_contig.data_ptr<float>();
+  const Tensor& input_contig =
+      input.is_contiguous() ? input : input.contiguous();
+  const float* input_ptr = input_contig.data_ptr<float>();
 
   TORCH_CHECK(input.dim() >= 2);
-  int64_t M = size_to_dim_(input.dim() - 1, input.sizes());
-  int64_t K = input.size(input.dim() - 1);
+  const int64_t M = size_to_dim_(input.dim() - 1, input.sizes());
+  const int64_t K = input.size(input.dim() - 1);
   TORCH_CHECK(weight.dim() == 2);
   TORCH_CHECK(K == weight.size(1));
-  auto N = weight.size(0);
+  const int64_t N = weight.size(0);
   TORCH_CHECK(bias.dim() == 1);
   TORCH_CHECK(bias.size(0) == N);
   TORCH_CHECK(weight_scale.isFloatingPoint());
   TORCH_CHECK(weight_zero_point.isIntegral(false));
 
   // Calculate statistics for quantization of the input Tensor
-  float x_min, x_max;
+  float x_min;
+  float x_max;
   fbgemm::FindMinMax(
-      /*m=*/input_ptr,
-      /*min=*/&x_min,
-      /*max=*/&x_max,
-      /*len=*/input.numel());
+      /* m = */ input_ptr,
+      /* min = */ &x_min,
+      /* max = */ &x_max,
+      /* len = */ input.numel());
 
   // Input tensor is quantized as 8-bit unsigned values
-  static constexpr int precision = 8;
-  static constexpr bool is_signed = false;
-  static constexpr int bound = (1 << (precision - 1));
+  constexpr int kPrecision = 8;
+  constexpr bool kIsSigned = false;
+  constexpr int kBound = (1 << (kPrecision - 1));
 
   // Calculate scale and zero point for quantization of input tensor
   auto q_params = fbgemm::ChooseQuantizationParams(
-      /*min=*/x_min,
-      /*max=*/x_max,
-      /*qmin=*/is_signed ? -bound : 0,
-      /*qmax=*/is_signed ? (bound - 1) : (1 << precision) - 1,
-      /*preserve_sparsity=*/false);
-
-  q_params.precision = precision;
+      /* min = */ x_min,
+      /* max = */ x_max,
+      /* qmin = */ kIsSigned ? -kBound : 0,
+      /* qmax = */ kIsSigned ? (kBound - 1) : (1 << kPrecision) - 1,
+      /* preserve_sparsity = */ false);
+  q_params.precision = kPrecision;
 
   // ReQuantizeForFloat requires pointers to the scale and zero point values,
   // since in the case of rowwise quantization these will be arrays rather than
   // scalars. But in this case, we're doing whole-tensor quantization so we just
   // pass a pointer to the scale values (and internally ReQuantizeFor Float
   // won't index past 0
-  float weight_scale_float = static_cast<float>(weight_scale.to<double>());
-  int32_t weight_zero_point_int32 =
+  const float weight_scale_float =
+      static_cast<float>(weight_scale.to<double>());
+  const int32_t weight_zero_point_int32 =
       static_cast<int32_t>(weight_zero_point.to<int64_t>());
 
-  auto bias_contig = bias.contiguous();
+  const auto& bias_contig = bias.is_contiguous() ? bias : bias.contiguous();
```

Comment:
Wonder if the `bias.contiguous();` implementation under the hood has got such checks?

Reply:
Actually the difference is if we use reference here, there will be no tensor shallow copy at all. But anyway we can change it back since it is not a serious thing.